server/src/test/java/org/apache/calcite/test/ServerTest.java

```diff
@@ -242,6 +242,19 @@ static Connection connect() throws SQLException {
     }
   }
 
+  @Test public void testInsertCreateNewCompositeUdt() throws Exception {
+    try (Connection c = connect();
+        Statement s = c.createStatement()) {
+      boolean b = s.execute("create type mytype as (i int, j int)");
+      assertThat(b, is(false));
+      b = s.execute("create table w (i int not null, j mytype)");
+      assertThat(b, is(false));
+      int x = s.executeUpdate("insert into w "
+          + "values (1, mytype(1, 1))");
+      assertThat(x, is(1));
```

Comment:
Can I add some insert table results to test?

Reply:
Done

Reply:
> Done :)

   What I mean is to use the following example to verify the results inserted into the table.

      final String select = "select * from w";
      try (ResultSet r = s.executeQuery(select)) {
        assertThat(r.next(), is(true));
        assertThat(r.getString(1), is("abc"));
        assertThat(r.getString(2), is("de "));
        assertThat(r.getInt(3), is(3));
        assertThat(r.getString(4), is("de"));
        assertThat(r.next(), is(false));
      }

Reply:
Yup, I have updated the test cases, for second column the type is Struct type.